cli/src/build.rs

```diff
@@ -147,7 +147,8 @@ pub fn build_project(arg_mode: &Option<String>, arg_platforms: &Option<Vec<Strin
     }
 
     // Architecture selection for iOS or Android
-    let selected_architectures = platform.select_archs();
+    let selected_architectures = platform.select_archs(&mut config);
+    write_config(&config_path, &config)?;
```

Comment:
Write the config file before build to avoid a failed build causes users have to select again.